components/compositing/webview_renderer.rs

```diff
@@ -333,18 +333,42 @@ impl WebViewRenderer {
 
         // If we can't find a pipeline to send this event to, we cannot continue.
         let get_pipeline_details = |pipeline_id| self.pipelines.get(&pipeline_id);
-        let result = match self
+        let result = self
             .global
             .borrow()
-            .hit_test_at_point(point, get_pipeline_details)
-        {
+            .hit_test_at_point(point, get_pipeline_details);
+        let result = match result {
             Ok(hit_test_results) => hit_test_results,
             Err(HitTestError::EpochMismatch) => {
                 self.pending_point_input_events
                     .borrow_mut()
                     .push_back(event);
                 return false;
             },
+            // If it didn't hit anything
+            Err(HitTestError::Others)
+                if matches!(
+                    event,
+                    InputEvent::MouseButton(_) | InputEvent::MouseMove(_) | InputEvent::Wheel(_)
+                ) =>
+            {
+                // Reset cursor to default
+                self.global
+                    .borrow_mut()
+                    .update_cursor(point, self.id, Some(Cursor::Default));
+                return if let Err(error) = self.global.borrow().constellation_sender.send(
+                    EmbedderToConstellationMessage::ForwardInputEvent(
+                        self.id,
+                        InputEvent::MouseLeave(MouseLeaveEvent::new(point)),
+                        None,
+                    ),
+                ) {
+                    warn!("Sending event to constellation failed ({error:?}).");
+                    false
+                } else {
+                    true
+                };
+            },
```

Comment:
Why is this necessary? Is the winit event sometimes never fired? Not hitting anything isn't quite the same as the mouse leaving the window.

Reply:
I thought it would be easier for the embedder that we do the hit test for them here guess we should leave that to the embedder